trainer/lib/trainer/options.rb

```diff
@@ -50,6 +50,11 @@ def self.available_options
                                      description: "Produces class name and test name identical to xcpretty naming in junit file",
                                      is_string: false,
                                      default_value: false),
+        FastlaneCore::ConfigItem.new(key: :force_legacy_xcresulttool,
```

Comment:
This new option key is going to help to rollout this implementation with minimized regression risk 👏